src/create-constructor-function.js

```diff
@@ -0,0 +1,56 @@
+/**
+ * Creates a constructor function from an ES2015 class, this is a workaround
+ * needed to being able to extend a class from code that's transpiled by Babel.
+ * See https://github.com/babel/babel/pull/8656
+ * @param {*} Type The ES2015 base class used to create the constructor
+ * @param {*} Parent The object where the prototype chain walk to copy over
+ * symbols and static properties to the constructor stops. If not provided,
+ * the chain stops at Object.
+ * @returns {Function} Constructor function than can be safely subclassed from
+ * transpiled code.
+ */
+function createConstructorFunction(Type, Parent) {
+	if (typeof Parent === "undefined") {
+		Parent = Object.getPrototypeOf(Object);
+	}
+
+	function TypeConstructor() {
+		return Reflect.construct(Type, arguments, this.constructor);
+	}
+
+	TypeConstructor.prototype = Object.create(Type.prototype);
+	TypeConstructor.prototype.constructor = TypeConstructor;
+
+	// Walk up the prototype chain to copy over all Symbols and
+	// static properties to the constructor function
+	let Link = Type;
+	while (Link !== Parent && Link !== null) {
+		const symbols = Object.getOwnPropertySymbols(Link);
+		const props = Object.getOwnPropertyNames(Link);
+
+		props.forEach(function(prop) {
+			if (!TypeConstructor[prop]) {
+				const value = Link[prop];
+				Object.defineProperty(TypeConstructor, prop, { value });
+			}
+		});
+
+		symbols.forEach(function(symbol) {
+			if (!TypeConstructor[symbol]) {
+				const value = Link[symbol];
+
+				Object.defineProperty(TypeConstructor, symbol, {
+					get() {
+						return symbol === Symbol.species ? this : value;
```

Comment:
Not sure why this check is needed, but can we do it outside of the getter so the check is not done on ever get? In other words...

```js
if(symbol === Symbol.species) {
  Object.defineProperty(TypeConstructor, symbol, {
    get() { return this; } // ...
  })
} else {
  Object.defineProperty( ...
}
```

Reply:
You can't just use the Symbol implementation on the parent type. It breaks `can-observable-array` filter test. That's why I had to make it return `this`. Instead of whatever the parent in the prototype chain is doing.

Reply:
I figured there was a reason, but can we check equality before defining the prop as shown above rather than checking equality on every get?

Reply:
I removed the whole conditional, seems what I was seeing was an artifact of not having the original property descriptor.